src/FindStmtCost.cpp

```diff
@@ -0,0 +1,930 @@
+            cost = NORMAL_NODE_DMC;
+        }
+
+        // this happens when visualizing cost of else_case in StmtToViz.cpp
+        else if (type == IRNodeType::Variable) {
+            const Variable *var = (const Variable *)node;
+            if (var->name == "canIgnoreVariableName") {
+                cost = NORMAL_NODE_DMC;
+            }
+        }
+
+        // else, error
+        else {
+            internal_error << "\n"
+                           << "FindStmtCost::get_data_movement_cost: " << print_node(node)
+                           << "node not found in stmt_cost map"
+                           << "\n\n";
+            return 0;
+        }
+    } else {
+        if (inclusive) cost = it->second.data_movement_cost_inclusive;
+        else
+            cost = it->second.data_movement_cost_exclusive;
+    }
+
+    if (cost < 0) {
+        internal_error << "\n"
+                       << "FindStmtCost::get_data_movement_cost: " << print_node(node)
+                       << "data_movement_cost_exclusive not set (cost is: " << cost << ")"
+                       << "\n\n";
+    }
+
+    return cost;
+}
+
+int FindStmtCost::get_if_node_cost(const IRNode *op, bool inclusive, bool is_computation) const {
+    if (op->node_type != IRNodeType::IfThenElse) {
+        internal_error << "\n"
+                       << "FindStmtCost::get_if_node_cost: " << print_node(op)
+                       << "node is not IfThenElse"
+                       << "\n\n";
+    }
+
+    int cost;
+    const IfThenElse *if_then_else = dynamic_cast<const IfThenElse *>(op);
+
+    if (is_computation) {
+        if (inclusive) {
+            int computation_cost = get_computation_cost(if_then_else->condition.get(), inclusive) +
+                                   get_computation_cost(if_then_else->then_case.get(), inclusive);
+            cost = computation_cost;
+        } else {
+            cost = NORMAL_NODE_CC;
+        }
+    } else {
+        if (inclusive) {
+            int data_movement_cost =
+                get_data_movement_cost(if_then_else->condition.get(), inclusive) +
+                get_data_movement_cost(if_then_else->then_case.get(), inclusive);
+            cost = data_movement_cost;
+        } else {
+            cost = NORMAL_NODE_DMC;
+        }
+    }
+    return cost;
+}
+
+vector<int> FindStmtCost::get_costs_children(const IRNode *parent, vector<const IRNode *> children,
+                                             bool inclusive) const {
+    int children_cc = 0;
+    int children_dmc = 0;
+
+    for (const IRNode *child : children) {
+        children_cc += get_computation_cost(child, inclusive);
+        children_dmc += get_data_movement_cost(child, inclusive);
+    }
+
+    vector<int> costs_children = {children_cc, children_dmc};
+
+    return costs_children;
+}
+
+void FindStmtCost::set_costs(
+    bool inclusive, const IRNode *node, vector<const IRNode *> children,
```

Comment:
Do you really want to pass this vector by value here? Wouldn't it be preferable to pass as a const ref?